plasma_framework/contracts/src/framework/ExitGameController.sol

```diff
@@ -0,0 +1,106 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "./models/ExitModel.sol";
+import "./interfaces/ExitProcessor.sol";
+import "./registries/ExitGameRegistry.sol";
+import "./priorityQueue/PriorityQueue.sol";
+
+contract ExitGameController is ExitGameRegistry {
+    uint64 public exitQueueNonce = 1;
+    mapping(uint256 => ExitModel.Exit) public exits;
+    mapping (address => PriorityQueue) public exitsQueues;
+
+    event TokenAdded(
+        address token
+    );
+
+    event ProcessedExitsNum(
+        uint256 processedNum,
+        address token
+    );
+
+    /**
+     * @notice Add token to the plasma framework and initiate the priority queue.
+     * @dev the queue is created as a new contract instance.
+     * @param _token Address of the token.
+     */
+    function addToken(address _token) external {
+        require(!hasToken(_token), "Such token has already been added");
+
+        address queueOwner = address(this);
+        exitsQueues[_token] = new PriorityQueue(queueOwner);
+        emit TokenAdded(_token);
+    }
+
+    /**
+     * @notice Checks if queue for particular token was created.
+     * @param _token Address of the token.
+     * @return bool represents whether the queue for a token was created.
+     */
+    function hasToken(address _token) public view returns (bool) {
+        return address(exitsQueues[_token]) != address(0);
+    }
+
+    /**
+     * @notice Enqueue exits from exit game contracts
+     * @dev Unique priority is a combination of original priority and an increasing nonce
+     * @dev Use public instead of external because structs in calldata not currently supported.
+     * @param _priority The priority of the exit itself
+     * @param _token Token for the exit
+     * @param _exit Exit data that contains the basic information for exit processor the process the exit
+     * @return a unique priority number computed for the exit
+     */
+    function enqueue(uint192 _priority, address _token, ExitModel.Exit memory _exit) public onlyFromExitGame returns (uint256) {
+        require(hasToken(_token), "Such token has not be added to the plasma framework yet");
+
+        PriorityQueue queue = exitsQueues[_token];
+        uint256 uniquePriority = (uint256(_priority) << 64 | exitQueueNonce);
```

Comment:
The idea here is that to make each ExitGame contract as independent as each others' logic as possible. With this nonce from the framework level, no matter the priority conflicts in two ExitGame contracts or not, it would work fine to the queue. Otherwise the ExitGame would need to check how all other contracts choose their priority and make sure the logic fits in.